includes/multi-currency/MultiCurrency.php

```diff
@@ -343,8 +331,8 @@ public function get_cached_currencies() {
 			return $cache_data;
```

Comment:
Everything is great, tests are covering and working for all scenarios, but on this part I had some doubt, If the cache is full of currencies, and we don't have connection to Stripe, should we still return the currencies? Or should we return an empty array?

Reply:
Ok found out that the cache has to do nothing with the available currency list, it only fills the missing values from the cache. The available currency list is already returned an empty array from the `get_account_available_currencies()` method, if Stripe isn't connected. 

Tested all scenarios you've written, and deleted all options related with MC from the database, tested again, and it didn't produce any fatal errors. The point that you've moved the stripe connection check seems to fix everything in this matter. I wasn't sure about the other assigned filters and checked them again, but couldn't find anything that can go wrong. New tests are also passing. Thanks for this fix! Looks good. 🚢 it.

Reply:
Ok found out that the cache has to do nothing with the available currency list, it only fills the missing values from the cache. The available currency list is already returned an empty array from the `get_account_available_currencies()` method, if Stripe isn't connected.